### PR TITLE
Add private key jwt guide

### PR DESCRIPTION
**Pull request by nmoskaleva**

### CODE REVIEW

File: src/pages/verify/guides/privatekey-jwt.mdx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    throw error;
+  }
+}
+```
+
+### Build the client assertion
+
+The client assertion is a JWT signed by the private key of the [RSA key pair](/verify/guides/privatekey-jwt/#generate-an-rsa-key-pair-and-jwks) generated earlier. It should contain the following claims:
+
+`alg`: The algorithm used to sign the assertion. Criipto Verify supports RSA256.
```
Comment (mickhansen) [inline src/pages/verify/guides/privatekey-jwt.mdx:69]: `alg` should be `RS256`.
I suppose we also support other asymmetric algorithms? @kasperhj @sgryt

File: src/pages/verify/guides/privatekey-jwt.mdx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+</Highlight>
+
+You can configure your [Criitp Verify Application](https://docs.criipto.com/verify/getting-started/basics/#applications) to use `private_key_jwt` for client authentication.
+
+## Authenticate with Private Key JWT
+
+To use `private_key_jwt` for client authentication, follow these three steps:
+
+1. Generate an RSA key pair and a JSON Web Key Set(JWKS).
```
Comment (kasperhj) [inline src/pages/verify/guides/privatekey-jwt.mdx:29]: Maybe these three steps are a bit superfluous, as they are each addressed in the subsequent sub-headings?

File: src/pages/verify/guides/privatekey-jwt.mdx (hunk 1 above)
Comment (kasperhj) [reply src/pages/verify/guides/privatekey-jwt.mdx]: @mickhansen This is left to the MS library to handle and with will probably suck and blow all kinds of algos.
The private key jwt client auth specs only requires an implementation RS256 which is consistent with what we expose in the OIDC metadata.

File: src/pages/verify/guides/privatekey-jwt.mdx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    throw error;
+  }
+}
+```
+
+### Exchange the client assertion for an access token
+
+You'll need to include the client assertion along with the verification code when performing the [code-for-token exchange](/verify/getting-started/oidc-intro/#exchange-the-code-for-a-token).
+
+The `client_assertion_type` should be set to `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`.
```
Comment (kasperhj) [inline src/pages/verify/guides/privatekey-jwt.mdx:117]: It MUST be set to this value.

File: src/pages/verify/guides/privatekey-jwt.mdx (hunk 1 above)
Comment (kasperhj) [inline src/pages/verify/guides/privatekey-jwt.mdx]: I suggest changing "sign the assertion" to "sign the JWT".
I think the special case is that this client auth scheme has a client assertion that happens to be a jwt, but in general I think all the principles about signing usually applies to jwt, but I might be mistaken.

File: src/pages/verify/guides/privatekey-jwt.mdx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+## Configure Private Key JWT Authentication
+
+To configure Private Key JWT Authentication, you'll need to register JWKS on behalf of your client on the [management dashboard](https://dashboard.criipto.com).
+
+In your Application settings, go to `OpenID Connect` > `Client JWKS`. You can add a static JWKS or a JWKS endpoint.
+
+![Client JWKS Endpoint](./images/JWKS_endpoint.png)
+
+- Static: Add your static JWKS.
+- JWKS Endpoint: Provide JWKS URI to fetch JWKS dynamically.
```
Comment (kasperhj) [inline src/pages/verify/guides/privatekey-jwt.mdx]: The URI in this case in a URL.

File: src/pages/verify/guides/privatekey-jwt.mdx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+## Advantages of Private Key JWT Authentication
+
+`private_key_jwt` offers enhanced security compared to traditional `client_secret` authentication methods by removing the need for a pre-shared symmetric key (client secret).
+
+Instead, you generate a pair of asymmetric keys and have Criipto point to the JWKS in your system. During authentication, your application signs its request with the private key, creating a signed client assertion. Criipto Verify can then use the public key from the JWKS to decrypt and verify the assertion.
+
+This setup also facilitates seamless key rollover.
```
Comment (kasperhj) [inline src/pages/verify/guides/privatekey-jwt.mdx:154]: I think "This setup" might be interpreted as `private_key_jwt`, but the seamless key rollover claim is only true when using the JWKS endpoint, not with the static URI.
Alternative off-the-top-of-my-head phrasing:
"When using `private_key_jwt` with a JWKS endpoint as opposed to a static JWKS you can benefit from seamless key rollover effectively avoiding change anything in the Criipto dashboard if you chose to rotate your keys."
Adjust appropriately if you find this helpful :)

**Review by kasperhj — COMMENTED** [review]
Nice.
A suggestion for future doc changes: it would be nice with a Diataxis context in which to review the changes. E.g. is it supposed to be a tutorial, etc.

File: src/pages/verify/guides/privatekey-jwt.mdx (hunk 5 above)
Comment (nmoskaleva) [reply src/pages/verify/guides/privatekey-jwt.mdx]: Very helpful. Thanks a lot @kasperhj. It's updated now.

**Comment by nmoskaleva** [comment]
**Diataxis: How-to guide**
(goal-oriented, guides the user’s action)

**Pull request merged**